lambdas/db-migration/src/migrations/20200911163145_create_providers_table.ts

```diff
@@ -0,0 +1,54 @@
+import * as Knex from 'knex';
+
+export const up = async (knex: Knex): Promise<void> =>
+  knex.schema.createTable('providers', (table) => {
+    table
+      .increments('cumulusId')
+      .primary();
+    table
+      .text('name')
+      .comment('Provider name')
+      .notNullable();
+    table
+      .enum('protocol', ['http', 'https', 'ftp', 'sftp', 's3'])
+      .comment('Protocol for the provider')
+      .notNullable();
+    table
+      .text('host')
+      .comment('Host name for the provider')
+      .notNullable();
+    table
+      .integer('port')
+      .comment('Port name for acessing the provider');
+    table
+      .text('username')
+      .comment('Username for acessing the provider');
+    table
+      .text('password')
+      .comment('password for acessing the provider');
+    table
+      .boolean('encrypted')
```

Comment:
Should this be `notNullable`?

Reply:
Hmm. It wasn't in the original schema, but I suppose if we're enforcing in the migration, it could be? Though we would also need to enforce it with the parallel writes